packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -469,6 +471,48 @@ export const run = async (context: $TSContext, resourceDefinition: $TSObject, re
     throw new AmplifyFault('DeploymentFault', {
       message: error.message,
     }, error);
+  } finally {
+    deleteBackendSnapshotVFS();
+  }
+};
+
+const takeBackendSnapshotVFS = (): void => {
```

Comment:
Given that this is already a pretty bloated file that has a lot of concerns, I'd consider refactoring all the snapshotting and VFS stuff into a standalone utility (will also make the unit tests more clear I think).